src/codegen/irgen/irgenerator.cpp

```diff
@@ -1814,6 +1814,7 @@ class IRGeneratorImpl : public IRGenerator {
         static const std::string newline_str("\n");
         static const std::string space_str(" ");
 
+        // TODO: why are we inline-generating all this code instead of just emitting a call to some runtime function?
```

Comment:
probably no good reason; as we've supported more of the print-statement-semantics we've just incrementally added them in.  it's a good question though if we should just switch it to calling a runtime function.